### PR TITLE
Remove unsafe code from System.Web.HttpUtility

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
There's no are label for _System.Web.HttpUtility_, so labeling it as _System.Net_ instead. This removes the remaining unsafe code from the _System.Web.HttpUtility_ assembly (except for the shared `HexEncoder.cs` file which is shared across most of our library code).

I didn't attempt to perform any code optimizations here or fix logical bugs. It's just replacing existing unsafe constructs with safer equivalents.

### CODE REVIEW

**Comment by GrabYourPitchforks** [comment]
Review is easier if you hide whitespace changes: https://github.com/dotnet/runtime/pull/43422/files?diff=split&w=1

**Review by wfurt — APPROVED** [review]
LGTM

**Comment by danmosemsft** [comment]
What is your perf strategy for such changes @GrabYourPitchforks ? Do you run benchmarks, or is it more of an eyeball thing?

**Comment by GrabYourPitchforks** [comment]
@danmosemsft More of an eyeball thing, plus a quick experiment in a standalone console app if I'm not entirely sure that a given code pattern will elide bounds checks. That's not really a problem here since the code is fairly straightforward, but if you look through https://github.com/dotnet/runtime/pull/43379 you'll see that some of the loop conditions have been subtly tweaked to result in better codegen.

**Comment by GrabYourPitchforks** [comment]
Unit test failures are known issue https://github.com/dotnet/runtime/pull/43422.

File: src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs [hunk 1]
```diff
@@ -73,44 +73,37 @@ private static unsafe void HtmlAttributeEncodeInternal(string s, TextWriter outp
...
-                    {
-                        output.Write(*pch++);
-                    }
+                output.Write(s.AsSpan(0, index));
 
-                    while (cch-- > 0)
+                ReadOnlySpan<char> remaining = s.AsSpan(index);
+                for (int i = 0; i < remaining.Length; i++)
+                {
+                    char ch = remaining[i];
```
Comment (stephentoub) [inline src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs:81]: FWIW, the above could also be simply:
```C#
foreach (char ch in s.AsSpan(index))
```
The compiler will rewrite that to be basically identical to what's above.

**Pull request merged**